### PR TITLE
Add deployment notes and rewrite readme

### PR DESCRIPTION
**Pull request by JakeChampion**
For easier reading: https://github.com/Financial-Times/polyfill-service/blob/documentation/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,51 +1,164 @@
...
+  * [Monitoring](#monitoring)
+  * [Library API Reference](#library-api-reference)
+  * [License](#license)
+
+
+Requirements
+------------
+
+Running Origami Build Service requires a few tools:
+  * [Git]: For downloading the source code
```
Comment (onishiweb) [inline README.md]: Git link not working

File: README.md [hunk 2]
```diff
@@ -1,51 +1,164 @@
...
+  * [Node.js] 5.x and [npm] 3.x: For installing the dependencies and running the application
```
Comment (j-mes) [inline README.md]: It might help to say that NPM gets installed during the Node installation?

File: README.md [hunk 3]
```diff
@@ -1,51 +1,164 @@
...
-## Installing as a service
 
-1. Install [git](http://git-scm.com/downloads) and [Node](http://nodejs.org) on your system
-2. Clone this repository to your system (`git clone git@github.com:Financial-Times/polyfill-service.git`)
-3. Run `npm install` (this will also run `grunt build` which you will need to do again if you edit any polyfills)
+Testing
+-------
 
-To run the app for **development**:
+The tests are split into for the service and tests for the polyfills. The polyfill tests require `SAUCE_USER_NAME` and `SAUCE_API_KEY` to be configured, view the[configuration](#configuration) section for more information.
```
Comment (j-mes) [inline README.md]: Put in a space between `the` `[configuration](#configuration)`

File: README.md [hunk 4]
```diff
@@ -1,51 +1,164 @@
...
+```
+
+Install the dependencies:
+```sh
+npm install
+```
+
+Build the polyfill sources and start the server, rebuiling and restarting whenever any changes are made to the project:
+```sh
+grunt dev
```
Comment (j-mes) [inline README.md]: Could mention somewhere that the server's name is `dev`. A lot of people have different ideas what the server name should be :)

File: README.md [hunk 5]
```diff
@@ -1,51 +1,164 @@
...
+  * `PORT`: The port on which to listen for HTTP requests (default 3000).
+  * `NODE_ENV`: Name of environment.  We use `dev`, `prod`, `ci` or `qa`.  Just used for logging.
+  * `FASTLY_SERVICE_ID`, `FASTLY_API_KEY`: Used to fetch and render cache hit stats on the [usage] page of the hosted documentation.  If not specified, no stats will be shown.
+  * `PINGDOM_CHECK_ID`, `PINGDOM_API_KEY`, `PINGDOM_ACCOUNT`, `PINGDOM_USERNAME`, `PINGDOM_PASSWORD`: Used to fetch and render uptime and response time stats on the [usage] page of the hosted documentation.  If not specified, no stats will be shown.
+  * `GRAPHITE_HOST`: Host to which to send Carbon metrics.  If not set, no metrics will be sent.
+  * `GRAPHITE_PORT`: Port on the `GRAPHITE_HOST` to which to send Carbon metrics (default 2002).
+  * `SAUCE_USER_NAME` and `SAUCE_API_KEY`: [Sauce Labs][sauce] credentials for grunt test tasks (not used by the service itself)
+  * `NODE_ENV`: Name of environment.  We use `dev`, `prod`, `ci` or `qa`.  Just used for logging.
```
Comment (onishiweb) [inline README.md]: Duplicated above

File: README.md [hunk 6]
```diff
@@ -1,51 +1,164 @@
...
+----------
+
+The [production][heroku-production] and [QA][heroku-qa] applications run on [Heroku].  The library is published to the public [npm] registry.
+
+Before creating a new deployment, update the compatability table:
+```sh
+grunt compatgen && git commit docs/assets/compat.json -m 'update compat.json'
+```
+
+We use [Semantic Versioning][semver] to tag releases.  Only tagged releases should hit production, this ensures that the `__about` endpoint is informative.  To tag a new release, use one of the following (this is the only time we allow a commit directly to `master`):
```
Comment (onishiweb) [inline README.md]: Semver link not working

**Comment by onishiweb** [comment]
Looks good. Seems fairly straight forward to follow. Couple of small issues with links and a duplicated config option, but all good apart from that.

**Comment by j-mes** [comment]
Ditto what Adam has said.

File: README.md (hunk 1 above)
Comment (JakeChampion) [inline README.md]: Fixed

File: README.md (hunk 4 above)
Comment (JakeChampion) [inline README.md]: `dev` is the name of the grunt task. Should we list out the grunt tasks somewhere?

File: package.json [hunk 7]
```diff
@@ -67,6 +67,7 @@
     "grunt-contrib-clean": "^0.6.0",
     "grunt-contrib-watch": "^0.6.1",
     "grunt-simple-mocha": "^0.4.0",
+    "haikro": "^2.3.2",
```
Comment (matthew-andrews) [inline package.json]: <3

File: README.md (hunk 4 above)
Comment (j-mes) [inline README.md]: I think if you have a list of grunt tasks, and with a brief description of what they do (if it's not already on Polyfill) then it probably would help too!

File: README.md (hunk 6 above)
Comment (JakeChampion) [inline README.md]: Fixed.

File: package.json (hunk 7 above)
Comment (JakeChampion) [inline package.json]: I found out it only works with Bash, had me running in circles wondering why my shell wouldn't work. Opening an issue for this :-)

Comment (JakeChampion) [inline package.json]: https://github.com/matthew-andrews/haikro/issues/100

Comment (matthew-andrews) [inline package.json]: as a workaround you could drop it into a `Makefile` with https://github.com/Financial-Times/n-makefile/blob/master/Makefile#L9 to force it to execute a bash shell even if you, as a developer, are using zsh.

File: README.md [hunk 8]
```diff
@@ -1,51 +1,163 @@
...
+
+Running Origami Build Service requires a few tools:
+  * [Git]: For downloading the source code
+  * [Node.js] 5.x and [npm] 3.x: For installing the dependencies and running the application (npm is installed with Node.js)
+  * [Grunt] 0.1.x: Used for automating tasks such as testing
+
+Running Locally
+---------------
+
+Clone the project to our system:
```
Comment (triblondon) [inline README.md]: Generally the README is now quite confused over the use of 'we' and 'our'.  Particularly here, where the system concerned is the personal environment of the reader, so 'our' makes it sound like the clone is being done in some kind of cloud service operated by the polyfill service maintainers.  I suggest using 'you' and 'your' to refer to the reader, and 'we' and 'our' to refer to the maintainers.

File: README.md [hunk 9]
```diff
@@ -1,51 +1,163 @@
...
+```
+
+Build the polyfill sources and start the server, rebuiling and restarting whenever any changes are made to the project:
+```sh
+grunt dev
+```
+
+
+Configuration
+-------------
```
Comment (triblondon) [inline README.md:61]: Probably worth using a consistent Markdown header syntax.

File: README.md [hunk 10]
```diff
@@ -1,51 +1,163 @@
...
-## Installing as a service
 
-1. Install [git](http://git-scm.com/downloads) and [Node](http://nodejs.org) on your system
-2. Clone this repository to your system (`git clone git@github.com:Financial-Times/polyfill-service.git`)
-3. Run `npm install` (this will also run `grunt build` which you will need to do again if you edit any polyfills)
+Testing
+-------
 
-To run the app for **development**:
+The tests are split into for the service and tests for the polyfills. The polyfill tests require `SAUCE_USER_NAME` and `SAUCE_API_KEY` to be configured, view the [configuration](#configuration) section for more information.
```
Comment (triblondon) [inline README.md]: "into for"

File: README.md [hunk 11]
```diff
@@ -1,51 +1,163 @@
...
 
-Run `grunt dev` from the root of the working tree.  This will watch your filesystem and automatically rebuild sources and restart if you make any changes to any of the app source code or polyfills.  If you change the docs, library or service (`docs`, `lib` and `service` directories) the service will be restarted.  If you change the polyfills or polyfill configs (`polyfills` directory), the polyfill sources will be recompiled before the service restarts, which makes the restart slightly slower.
+```sh
+grunt test           # run service tests and polyfill tests on a small set of browsers
+grunt simplemocha    # run the service tests
+grunt ci             # run the service tests and polyfills tests on a large set of browsers
+```
+
+We run the tests on CI, we can see the [results on CircleCI][ci].  `grunt ci` must pass before we merge a pull request.
```
Comment (triblondon) [inline README.md]: Continuous integration testing runs on CircleCI.

File: README.md [hunk 12]
```diff
@@ -1,51 +1,163 @@
...
+----------
+
+The [production][heroku-production] and [QA][heroku-qa] applications run on [Heroku].  The library is published to the public [npm] registry.
+
+Before creating a new deployment, update the compatability table:
+```sh
+grunt compatgen && git commit docs/assets/compat.json -m 'update compat.json'
+```
+
+We use [Semantic Versioning][semver] to tag releases.  Only tagged releases should hit production, this ensures that the `__about` endpoint is informative.  To tag a new release, use one of the following (this is the only time we allow a commit directly to `master`):
```
Comment (triblondon) [inline README.md]: "Only tagged releases should hit production, _which_ ensures".

Also `/__about`

File: README.md [hunk 13]
```diff
@@ -119,10 +229,25 @@ Example:
 require('polyfill-service').listAllPolyfills();
 ```
 
-## cdn.polyfill.io deployment
+License
+-------
+
+The Financial Times has published this software under the [CC0 license][license].
```
Comment (triblondon) [inline README.md]: We now publish everything under MIT, so we should update this.  CC0 work can be re-licensed as MIT so the whole thing can be treated as MIT, except specific third party polyfills.  Also we still use the license so it should be present tense.  I suggest:

"Except where indicated in selected polyfill config files, the polyfill service codebase is licensed under the terms of the [MIT license].  Contributors must accept our [contribution terms].

File: package.json (hunk 7 above)
Comment (triblondon) [inline package.json]: I have a branch replacing haikro with docker, but I'm confused about Docker.

**Comment by triblondon** [comment]
Apart from nitpicking comments, this is really excellent.

File: README.md (hunk 11 above)
Comment (JakeChampion) [inline README.md]: That's what I was aiming to explain by the previous sentence.

> We run the tests on CI, we can see the [results on CircleCI][ci].

How should that be reworded to better explain this?

Comment (j-mes) [inline README.md]: _The continuous integration tests are run on Circle CI. We can see the results afterwards._

How about that?

**Comment by JakeChampion** [comment]
Updated readme, ready for another review @triblondon

**Comment by triblondon** [comment]
This is fine, thanks :+1:

**Pull request merged**